### PR TITLE
port some minibuffer functions

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/minibuf.rs [hunk 1]
```diff
@@ -65,4 +69,51 @@ pub fn minibuffer_prompt() -> LispObject {
...
+
+    let end = field_end(Some(beg), false, None);
+    let buffer_end = ThreadState::current_buffer().zv as EmacsInt;
+    if end == buffer_end && get_char_property(beg, Qfield, LispObject::constant_nil()).is_nil() {
+        beg
+    } else {
+        end
+    }
+}
+
```
Comment (shaleh) [inline rust_src/src/minibuf.rs]: Do it makes sense to collapse these two into an internal function? The only difference is the true/false for adding text properties. I know the C did not do it. Should we?

**Comment by shaleh** [comment]
Looks good. Are there tests hitting it?

File: rust_src/src/minibuf.rs (hunk 1 above)
Comment (brotzeit) [reply rust_src/src/minibuf.rs]: Hmmm. I'm not sure if this is worth an additional function.

**Comment by brotzeit** [comment]
I don't know how to write tests for functions that open a prompt.

**Pull request merged**